src/modules/Dashboard/features/RespondentData/RespondentDataReview/RespondentDataReview.tsx

```diff
@@ -251,7 +254,7 @@ export const RespondentDataReview = () => {
 
     handleSetInitialDate(new Date());
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [lastActivityCompleted]);
+  }, [lastActivityCompleted, selectedActivity?.id]);
```

Comment:
I'm assuming you discovered you needed to trigger the effect when the activity ID becomes available, but shouldn't that also be true for the flow ID?

```suggestion
  }, [lastActivityCompleted, selectedActivity?.id, selectedFlow?.id]);
```

Reply:
🎯 @farmerpaul